src/types.ts

```diff
@@ -16,6 +16,16 @@ export type CaptureCallback = (response: any, data: any) => void
 export type AutocaptureCompatibleElement = 'a' | 'button' | 'form' | 'input' | 'select' | 'textarea' | 'label'
 export type DomAutocaptureEvents = 'click' | 'change' | 'submit'
 
+export interface EventCaptureConfig {
+    /**
+     * An object mapping events to URLs to allow capture on, can be strings to match
+     * or regexes e.g. ['https://example.com', 'test.com/.*']
+     *
+     * Events that are not in this object will be allowed on all URLs
+     */
+    url_allowlists?: Record<string, (string | RegExp)[]>
+}
```

Comment:
Why do we need a separate config object with just one parameter?

That's like "just in case" planning for the future when your app will also support 3d rendering, variable width fonts and quantum entanglement. Let's build for what we have now...